common/services/opening-times.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+// TODO - capture opening hours in Prismic, then this can become part of prismic services
+import {placesOpeningHours} from '../model/opening-hours';
+import {isDatePast} from '../utils/format-date';
+import groupBy from 'lodash.groupby';
+import moment from 'moment';
+import type {ExceptionalVenueHours} from '../model/opening-hours';
+
+function london(d) {
+  // $FlowFixMe
+  return moment.tz(d, 'Europe/London');
+};
+
+function exceptionalOpeningDates(placesHoursArray): Array<?Date> {
+  return [].concat.apply([], placesHoursArray.map(place => {
+    return place.openingHours.exceptional &&
+      place.openingHours.exceptional.map(exceptionalDate => exceptionalDate.overrideDate);
+  }).filter(_ => _))
+    .sort((a, b) => Number(a) - Number(b))
+    .filter((item, i, array) => {
+      const firstDate = item;
+      const prevDate = array[i - 1];
+      if (!i) {
+        return true;
+      } else if (firstDate instanceof Date && prevDate instanceof Date) {
```

Comment:
If the type is truly `Date[]` we should just be able to `placesHoursArray.filter(Boolean).map(date: Date => doStuff)`.